app/models/accounting/quickbooks/transaction_creator.rb

```diff
@@ -78,6 +85,17 @@ def create_customer_reference(qb_customer_id:)
         entity
       end
 
+      def find_or_create_qb_customer_id_from(organization:)
+        return organization.qb_id if organization.qb_id.present?
+
+        qb_customer = ::Quickbooks::Model::Customer.new
+        qb_customer.display_name = organization.name
+
+        new_qb_customer = customer_service.create(qb_customer)
+
+        new_qb_customer.id
+      end
+
```

Comment:
This is feeling a bit bloat-y to me, i.e. violation of SRP. Can we make a new class `Customer' that is initialized with an Organization and has public method `reference`, which calls private method `find_or_create...`?

Then anywhere you need a customer reference you do `Customer.new(organization).reference`?

... and don't forget the class description comment :)